backend/src/main/java/com/morak/back/auth/ui/AuthResolver.java

```diff
@@ -11,22 +13,34 @@
 
 public class AuthResolver implements HandlerMethodArgumentResolver {
 
+    private final TokenProvider tokenProvider;
+
+    public AuthResolver(TokenProvider tokenProvider) {
+        this.tokenProvider = tokenProvider;
+    }
 
     @Override
     public boolean supportsParameter(MethodParameter parameter) {
         return parameter.hasParameterAnnotation(Auth.class);
     }
 
     @Override
-    public Object resolveArgument(MethodParameter parameter,
+    public Long resolveArgument(MethodParameter parameter,
                                   ModelAndViewContainer mavContainer,
                                   NativeWebRequest webRequest,
                                   WebDataBinderFactory binderFactory) {
         HttpServletRequest request = (HttpServletRequest) webRequest.getNativeRequest();
-        Long memberId = (Long) request.getAttribute("memberId");
-        if (memberId == null) {
-            throw new AuthorizationException("리졸버에서 멤버 ID를 가져오는데 실패했습니다.");
+
+        String token = getToken(request);
+        String payload = tokenProvider.getPayload(token);
+        return Long.parseLong(payload);
+    }
+
+    private String getToken(HttpServletRequest request) {
```

Comment:
```suggestion
    private String extractToken(HttpServletRequest request) {
```

getter 대신 정확한 용어를 사용하는게 좋을 것 같아요 ~

Reply:
그렇네요 .. ㅎ